### ISSUE TITLE
Validation loss overfitting

### ISSUE THREAD

**Issue by xbzzgogogo** [Caiyq2019/NeuralMAG#1]
Hello,
I want to reproduce your excellent paper, but i have trouble  
<img width="986" height="739" alt="Image" src="https://github.com/user-attachments/assets/c148020c-2fc9-431f-bc13-d94f5b0272a8" />
my main venv as follows
  pillow == 12.0.0
  python == 3.13.5
  torch == 2.7.1+cu118
  numpy == 2.2.6
  matplotlib == 3.10.0
  opencv-python == 4.12.0.88
  tqdm == 4.67.1
At first, i run the run.sh to generate the datasets using RTX 4060, but when i checked the background process, i found the committed memory  was approximately 5 times larger than my default value 32G. When i run the run.sh, it crashed some times, then i reset the parameters in run.sh. I just calculate the 32 and 64 size one time, and then i continue to calculate the remaining part. Dose that matter? I don't know. I used the gen-data to train on server, then i get the figure. 
Can someone help me ? Thank you in advance.

**Comment by Haroldpba** [comment]
Thank you for your question and for providing the environment information. From your description, I understand that you ran `run.sh` to generate the dataset, and occasional crashes during dataset generation could indeed be related to memory allocation, depending on how many samples or resolutions are processed simultaneously.

However, I am not completely sure what you mean by:

> *“I just calculate the 32 and 64 size one time, and then I continue to calculate the remaining part.”*

If possible, it would help to share:

* the modified `run.sh`
* the dataset directory structure

This will make it easier to determine whether the behavior you observed is expected or caused by a configuration issue.

Thank you again for your interest in the project and for taking the time to report this issue. Your feedback and support are greatly appreciated.

**Comment by xbzzgogogo** [comment]
> Thank you for your question and for providing the environment information. From your description, I understand that you ran `run.sh` to generate the dataset, and occasional crashes during dataset generation could indeed be related to memory allocation, depending on how many samples or resolutions are processed simultaneously.
> 
> However, I am not completely sure what you mean by:
> 
> > _“I just calculate the 32 and 64 size one time, and then I continue to calculate the remaining part.”_
> 
> If possible, it would help to share:
> 
> * the modified `run.sh`
> * the dataset directory structure
> 
> This will make it easier to determine whether the behavior you observed is expected or caused by a configuration issue.
> 
> Thank you again for your interest in the project and for taking the time to report this issue. Your feedback and support are greatly appreciated.

[run.sh](https://github.com/user-attachments/files/25868430/run.sh)
Thank you very much for your  response!

First of all, I want to apologize for my poor English. Please bear with me if my explanation is not clear enough.  I'll do my best to describe my issue.

As you can see from my run.sh content, I only changed the value of `w` in the first loop. I also modified the run.sh  to generate only 96x96 and modified the run.sh again to generate only 128x128

Here is a screenshot of my dataset directory structure as you requested:

<img width="471" height="279" alt="Image" src="https://github.com/user-attachments/assets/b793477f-ee65-4ddd-8b95-9e9d0a2b6651" />

Thank you again for taking the time to look into this issue！

**Comment by Haroldpba** [comment]
Thanks for the clarification and for sharing the dataset structure.

From what you described, splitting the data generation into multiple runs should generally not be an issue by itself, as long as all subsets are generated with consistent settings. In my case, I was able to train the model normally on a locally generated dataset without encountering this kind of generalization issue, so it is likely that something in the data distribution or training configuration is different.

One possible reason I’m considering is that, due to storage or memory constraints, the effective training dataset size or composition might have been reduced or altered (for example, fewer samples at certain resolutions, or an imbalanced mix of scales). This could affect the overall data distribution and lead to weaker generalization.

To better understand what’s going on, could you share your training script (especially the modifications you made)?

With that information, it should be easier to pinpoint the cause.

Thanks!

**Comment by Haroldpba** [comment]
For reference, here is part of the dataset loading output from my side:

<img width="1858" height="946" alt="Image" src="https://github.com/user-attachments/assets/cf17b38f-9c06-416b-948a-d5aff4cdd228" />

**Comment by xbzzgogogo** [comment]
> Thanks for the clarification and for sharing the dataset structure.
> 
> From what you described, splitting the data generation into multiple runs should generally not be an issue by itself, as long as all subsets are generated with consistent settings. In my case, I was able to train the model normally on a locally generated dataset without encountering this kind of generalization issue, so it is likely that something in the data distribution or training configuration is different.
> 
> One possible reason I’m considering is that, due to storage or memory constraints, the effective training dataset size or composition might have been reduced or altered (for example, fewer samples at certain resolutions, or an imbalanced mix of scales). This could affect the overall data distribution and lead to weaker generalization.
> 
> To better understand what’s going on, could you share your training script (especially the modifications you made)?
> 
> With that information, it should be easier to pinpoint the cause.
> 
> Thanks!

Thank you for your response!

Here is my run_train.sh and train.py, I didn't modify other function files in the train directory.

[run_train.sh](https://github.com/user-attachments/files/26776923/run_train.sh)

[train.py](https://github.com/user-attachments/files/26777500/train.py)

Actually, I reran your work on a supercomputing platform using an RTX 4090D. However, due to platform limitations, I used CUDA 11.8 and Python 3.11. It was also quite some time ago, so I can no longer find the specific versions of the libraries used in the conda environment. I only kept the resulting figures.Here is the loss figure:

<img width="638" height="479" alt="Image" src="https://github.com/user-attachments/assets/7a3aebdc-4cdf-4ae6-9c8f-a44c7045afde" />

I suspect that the difference may be due to the CUDA and Python versions. The loss curve obtained this time seems relatively more normal.
